### PR TITLE
build(plugin-keychain-memory): fix local imports broken due to .js exts

### LINKED ISSUE #3254: build(plugin-keychain-memory): fix local imports broken due to .js extensions
### Description

Reproduce by executing this on `main`:

`yarn jest packages/cactus-plugin-keychain-memory/src/test/typescript/integration/api-surface.test.ts`
and then watch it fail as such:

```sh
$ yarn jest packages/cactus-plugin-keychain-memory/src/test/typescript/integration/api-surface.test.ts
jest-haste-map: duplicate manual mock found: ConfigUtil
  The following files share their name; please delete one of them:
    * <rootDir>/packages/cactus-cmd-socketio-server/src/main/typescript/routing-interface/util/__mocks__/ConfigUtil.ts
    * <rootDir>/packages/cactus-cmd-socketio-server/dist/lib/main/typescript/routing-interface/util/__mocks__/ConfigUtil.js

 FAIL  packages/cactus-plugin-keychain-memory/src/test/typescript/integration/api-surface.test.ts
  ● Test suite failed to run

    Cannot find module './default_service_pb.js' from 'packages/cactus-plugin-keychain-memory/src/main/typescript/generated/crpc/services/default_service_connect.ts'

    Require stack:
      packages/cactus-plugin-keychain-memory/src/main/typescript/generated/crpc/services/default_service_connect.ts
      packages/cactus-plugin-keychain-memory/src/main/typescript/plugin-keychain-memory.ts
      packages/cactus-plugin-keychain-memory/src/main/typescript/public-api.ts
      packages/cactus-plugin-keychain-memory/src/test/typescript/integration/api-surface.test.ts

      13 | // @ts-nocheck
      14 |
    > 15 | import { DeleteKeychainEntryV1Request, GetKeychainEntryV1Request, GetPrometheusMetricsV1Response, HasKeychainEntryV1Request, SetKeychainEntryV1Request } from "./default_service_pb.js";
         | ^
      16 | import { DeleteKeychainEntryResponseV1PB } from "../models/delete_keychain_entry_response_v1_pb_pb.js";
      17 | import { Empty, MethodKind } from "@bufbuild/protobuf";
      18 | import { GetKeychainEntryResponseV1PB } from "../models/get_keychain_entry_response_v1_pb_pb.js";

      at Resolver._throwModNotFoundError (node_modules/jest-resolve/build/resolver.js:427:11)
      at Object.<anonymous> (packages/cactus-plugin-keychain-memory/src/main/typescript/generated/crpc/services/default_service_connect.ts:15:1)
      at Object.<anonymous> (packages/cactus-plugin-keychain-memory/src/main/typescript/plugin-keychain-memory.ts:26:1)
      at Object.<anonymous> (packages/cactus-plugin-keychain-memory/src/main/typescript/public-api.ts:4:1)
      at Object.<anonymous> (packages/cactus-plugin-keychain-memory/src/test/typescript/integration/api-surface.test.ts:1:1)

Test Suites: 1 failed, 1 total
Tests:       0 total
Snapshots:   0 total
Time:        0.61 s
Ran all test suites matching /packages\/cactus-plugin-keychain-memory\/src\/test\/typescript\/integration\/api-surface.test.ts/i.
```

### Acceptance Criteria
1. Fix the generated code which appears to be importing `.js` extension files **OR** fix the Jest config whichever turns out to be the solution that doesn't add more tech. debt.

### PR DESCRIPTION
**Pull request by petermetz**
Found the solution and a long discussion about it here:
https://github.com/kulshekhar/ts-jest/issues/1057

TLDR: The Jest resolver needs a little extra information/tweak to the
config so that it can correctly handle the .js imports.

Specifically this comment provided the solution which I made here:
https://github.com/kulshekhar/ts-jest/issues/1057#issuecomment-1482644543

Fixes #3254

Signed-off-by: Peter Somogyvari <peter.somogyvari@accenture.com>

**Pull Request Requirements**
- [x] Rebased onto `upstream/main` branch and squashed into single commit to help maintainers review it more efficient and to avoid spaghetti git commit graphs that obfuscate which commit did exactly what change, when and, why.
- [x] Have git sign off at the end of commit message to avoid being marked red. You can add `-s` flag when using `git commit` command. You may refer to this [link](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) for more information.
- [x] Follow the Commit Linting specification. You may refer to this [link](https://www.conventionalcommits.org/en/v1.0.0-beta.4/#specification) for more information. 

**Character Limit**
- [x] Pull Request Title and Commit Subject must not exceed 72 characters (including spaces and special characters).
- [x] Commit Message per line must not exceed 80 characters (including spaces and special characters).

**A Must Read for Beginners**
For rebasing and squashing, here's a [must read guide](https://github.com/servo/servo/wiki/Beginner's-guide-to-rebasing-and-squashing) for beginners.

### CODE REVIEW

**Review by outSH — APPROVED** [review]

**Review by jagpreetsinghsasan — APPROVED** [review]

**Review by sandeepnRES — APPROVED** [review]

**Comment by sandeepnRES** [comment]
Hi @petermetz `yarn-codegen` test is failing. As it is a required test, hence I approved the PR, as it is already blocking the PR.

**Comment by petermetz** [comment]
> Hi @petermetz `yarn-codegen` test is failing. As it is a required test, hence I approved the PR, as it is already blocking the PR.

@sandeepnRES Thank you for the review! I'm retrying the codegen task now (it's the old flake that has returned so it usually just needs another go, fingers crossed)
